plugins/arcaniax_block_exporter.js

```diff
@@ -0,0 +1,450 @@
+		var models = rp + "\\" + "models";
+		var blockModels = rp + "\\" + "models" + "\\" + "blocks";
+		var blockTextures = rp + "\\textures\\" + "blocks";
+		
+		makeFolder(bp);
+		makeFolder(rp);
+		makeFolder(blockBehaviors);
+		makeFolder(lang);
+		makeFolder(textures);
+		makeFolder(blockTextures);
+		makeFolder(models);
+		makeFolder(blockModels);
+		
+		createManifests(bp, rp, formData);
+		createBehaviorFile(blockBehaviors, formData);
+		saveModel(blockModels, formData);
+		saveTexture(blockTextures, formData);
+		addTranslation(lang, formData);
+		addTextureMapping(textures, formData);
+		
+		Blockbench.showQuickMessage('Block was exported successfully!', 5000);
+	}
+	
+	function makeFolder(dir){
+		if (!fs.existsSync(dir)){
+			fs.mkdirSync(dir);
+		}
+	}
+	
+	function saveModel(dir, formData){
+		fs.writeFile(dir + "\\" + formData.file_name +".geo.json", Format.codec.compile() , onError);
+	}
+	
+	function createManifests(bp, rp, formData){
+		fs.writeFile(bp + "\\manifest.json", getBehaviorManifest(formData) , onError);
+		fs.writeFile(rp + "\\manifest.json", getResourceManifest(formData) , onError);
+	}
+	
+	function addTranslation(dir, formData){
+		var translationPrefix = "tile." + formData.id + ".name=";
+		var translation = translationPrefix + formData.name;
+		
+		fs.writeFile(dir + "\\en_US.lang", translation , onError);
+		fs.writeFile(dir + "\\languages.lang", getDefaultLanguages() , onError);
+	}
+	
+	function getDefaultLanguages(){
+		return `["en_US"]`;
+	}
+	
+	function addTextureMapping(dir, formData){
+		fs.writeFile(dir + "\\terrain_texture.json", getTerrainTextureFormat(formData) , onError);
+	}
+	
+	function saveTexture(dir, formData){
+		if (textures[0]){
+			var base64 = textures[0].getBase64();
+			var base64Data = base64.replace(/^data:image\/png;base64,/, "");
+
+			fs.writeFile(dir + "\\" + formData.file_name + ".png", base64Data, 'base64', onError );
+		}
+	}
+	
+	function createBehaviorFile(dir, formData){
+		fs.writeFile(dir + "\\" + formData.file_name + '.behavior.json', getBehaviorFormat(formData) , onError);
+	}
+	
+	function checkFolders(dir){
+		var bp = dir+"\\behavior_packs";
+		var rp = dir+"\\resource_packs";
+		if (fs.existsSync(bp)&&fs.existsSync(rp)) {
+			if (hasDirectory(rp)){
+				if (hasDirectory(bp)){
+				  return true;
+				}
+			}
+		}
+		return false;
+	}
+	
+	function hasDirectory(dir){
+		var bool = false;
+		fs.readdirSync(dir).forEach(file => {
+		  if (fs.lstatSync(dir + "\\" + file).isDirectory() ){
+			  bool = true;
+		  }
+		});
+		return bool;
+	}
+	
+	function getFirstDirectory(dir){
+		var returnFile = "";
+		fs.readdirSync(dir).forEach(file => {
+		  if (fs.lstatSync(dir + "\\" + file).isDirectory() ){
+			  returnFile = file;
+		  }
+		});
+		return dir + "\\" + returnFile;
+	}
+	
+	
+	function getResourceManifest(formData){
+		var name = formData.name;
+		var uuid1 = generateUUID();
+		var uuid2 = generateUUID();
+		var json =
+`{
+    "format_version": 2,
```

Comment:
Can't help but notice that the JSON files all use a different type of indentation. Maybe use a consistent method of indentation, like only tabs.
You could also use the built-in function `compileJSON()` to generate the JSON strings, then you won't need to worry about indentation and you can also write the JSON code directly in code, without having to deal with template strings.